### ISSUE TITLE
Reify causes node.js to crash when run from cli (0.8.x only)

### ISSUE THREAD

**Issue by cmawhorter** [benjamn/reify#153]
1. Create the two files described below
2. `node ./test.js`
3. Node.js crashes due to a failed assertion

```
../src/async-wrap-inl.h:38:v8::Local<v8::Value> node::AsyncWrap::MakeCallback(const v8::Local<v8::String>, int, v8::Local<v8::Value> *): Assertion `cb_v->IsFunction()' failed.
```

mac 10.12.4; nodejs 7 (and seemingly 6; possible others)

index.js
```js
console.log('pre');
require('reify');
const imp = require('./test-es6.js').default;
console.log('post');
setTimeout(() => console.log('1s'), 1000); // never reached
```

test-es6.js
```js
export default {};
```

Edit:  Oops.  Forgot to mention I could repro in reify `0.8.0`, `.2`, and `.3`. `@0.7.x` did not have problem.

**Comment by jdalton** [comment]
Double check your node version.
```
node -v
```

This is an error from the internal stat module which should be guarded against for supported node versions 4,6,7.

**Comment by cmawhorter** [comment]
just confirmed with node versions:

- 4.3.2
- 6.3.1 
- 7.4.0

All crashed but 4 gives a seemingly more useful error:

```
/Users/...snip.../node_modules/reify/node_modules/minizlib/index.js:136
    this[_buffer] = Buffer.allocUnsafe(this[_chunkSize])
                           ^

TypeError: Buffer.allocUnsafe is not a function
    at Zlib (/Users/...snip.../node_modules/reify/node_modules/minizlib/index.js:136:28)
    at Gzip (/Users/...snip.../node_modules/reify/node_modules/minizlib/index.js:297:10)
    at gzip (/Users/...snip.../node_modules/reify/node/utils.js:142:25)
    at /Users/...snip.../node_modules/reify/node/utils.js:319:31
    at Array.forEach (native)
    at Immediate._onImmediate (/Users/...snip.../node_modules/reify/node/utils.js:315:32)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
$ node -v
v4.3.2
```

**Comment by jdalton** [comment]
Ah interesting. It happens to pass on the ones we test in travis. I can repro this. It's a bug in `minizlib`.

**Comment by jdalton** [comment]
Ah not `minizlib`... I'll have a patch in a couple minutes.

**Comment by benjamn** [comment]
This has been released in `reify@0.8.4`!
